RevenueCatUI/PaywallView.swift

```diff
@@ -6,52 +6,145 @@ import SwiftUI
     private let introEligibility: TrialOrIntroEligibilityChecker?
     private let purchaseHandler: PurchaseHandler?
 
-    /// Create a view for the given offering and paywal.
+    @State
+    private var offering: Offering?
+    @State
+    private var paywall: PaywallData?
+
+    /// Create a view that loads the `Offerings.current`.
     /// - Warning: `Purchases` must have been configured prior to displaying it.
-    public init(mode: PaywallViewMode, offering: Offering, paywall: PaywallData) {
+    public init(mode: PaywallViewMode = .default) {
         self.init(
+            offering: nil,
+            paywall: nil,
             mode: mode,
+            introEligibility: Purchases.isConfigured ? .init() : nil,
+            purchaseHandler: Purchases.isConfigured ? .init() : nil
+        )
+    }
+
+    /// Create a view for the given offering and paywal.
+    /// - Warning: `Purchases` must have been configured prior to displaying it.
+    public init(offering: Offering,
+                paywall: PaywallData,
+                mode: PaywallViewMode = .default) {
+        self.init(
             offering: offering,
             paywall: paywall,
+            mode: mode,
             introEligibility: Purchases.isConfigured ? .init() : nil,
             purchaseHandler: Purchases.isConfigured ? .init() : nil
         )
     }
 
     init(
-        mode: PaywallViewMode = .fullScreen,
-        offering: Offering,
-        paywall: PaywallData,
+        offering: Offering?,
+        paywall: PaywallData?,
+        mode: PaywallViewMode = .default,
         introEligibility: TrialOrIntroEligibilityChecker?,
         purchaseHandler: PurchaseHandler?
     ) {
-        self.mode = mode
-        self.offering = offering
-        self.paywall = paywall
+        self._offering = .init(initialValue: offering)
+        self._paywall = .init(initialValue: paywall)
         self.introEligibility = introEligibility
         self.purchaseHandler = purchaseHandler
+        self.mode = mode
     }
 
     // swiftlint:disable:next missing_docs
     public var body: some View {
         if let checker = self.introEligibility, let purchaseHandler = self.purchaseHandler {
-            self.paywall
-                .createView(for: self.offering, mode: self.mode)
-                .environmentObject(checker)
-                .environmentObject(purchaseHandler)
+            if let offering = self.offering {
+                if let paywall = self.paywall {
+                    LoadedOfferingPaywallView(
+                        offering: offering,
+                        paywall: paywall,
+                        mode: mode,
+                        introEligibility: checker,
+                        purchaseHandler: purchaseHandler
+                    )
+                } else {
+                    DebugErrorView("Offering '\(offering.identifier)' has no configured paywall",
+                                   releaseBehavior: .emptyView)
```

Comment:
For cases when developers do this:
```swift
PaywallView(mode: .banner)
```

But the offering doesn't have any paywalls.
In debug builds this will show a red banner with an error.
In release builds, it'll just be empty.